components/kyma-environment-broker/internal/process/provisioning/inject_btp_operator_credentials_step.go

```diff
@@ -129,14 +129,25 @@ func updateSecretData(secret *apicorev1.Secret, secretFromParameters *apicorev1.
 
 func (s *InjectBTPOperatorCredentialsStep) createOrRetry(k8sClient client.Client, newSecret *apicorev1.Secret, err error, log logrus.FieldLogger) error {
 	if apierrors.IsNotFound(err) {
+		namespace := &apicorev1.Namespace{ObjectMeta: v1.ObjectMeta{Name: secretNamespace}}
+		err = k8sClient.Create(context.Background(), namespace)
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
```

Comment:
I would check if err is different from AlreadyExists error then use log to provide extra info with details why Create() method failed. When the error is AlreadyExists error, skip the log as the required namespace exists (we don't care who/what created it) and try to create the Secret in existing namespace.

Reply:
Considering we will use the same step for update/upgrade you are right.

Reply:
changed